src/public/app/player/player.component.ts

```diff
@@ -25,20 +23,23 @@ const template: string = `
     template,
     selector: 'player',
 })
-export class Player implements OnChanges {
-
-    @Input() private song: Song;
+class Player implements OnChanges {
 
+    @Input() public queuedSong: QueuedUpSong;
+    @Output() public songFinished: EventEmitter = new EventEmitter();
     private player: ScPlayer;
```

Comment:
I think instead of putting an ScPlayer on the Player object, we should have the Song interface have a getPlayer function. That way we don't have to do

if (song.isSoundcloud) {
  scPlayer.play(song);
} else if (song.isSpotify) {
  spotifyPlayer.play(song);
}